### PR TITLE
EVG-2306: make request context load data lazily

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: plugin/plugin_ui.go [hunk 1]
```diff
@@ -88,14 +84,10 @@ type UIPanel struct {
 
 // UIContext stores all relevant models for a plugin page.
 type UIContext struct {
-	Settings   evergreen.Settings
```
Comment (john-m-liu) [inline plugin/plugin_ui.go]: plugin/plugin_ui_test.go:418:20: ctx.Task undefined (type UIContext has no field or method Task)
plugin/plugin_ui_test.go:418:34: ctx.Build undefined (type UIContext has no field or method Build)
plugin/plugin_ui_test.go:418:49: ctx.Version undefined (type UIContext has no field or method Version)

File: model/context.go [hunk 2]
```diff
@@ -13,114 +13,219 @@ import (
 // function, all the other applicable fields in the Context can
 // inferred and populated from the id of any one of the fields.
 type Context struct {
-	Task       *task.Task
-	Build      *build.Build
-	Version    *version.Version
-	Patch      *patch.Patch
-	ProjectRef *ProjectRef
+	projectRef *ProjectRef
```
Comment (john-m-liu) [inline model/context.go]: Line 60 (and possibly others) of base_angular.html also need to be changed since you changed the casing of the context variables

**Comment by tychoish** [comment]
I've pushed a couple of patches that address the plugin ui tests and the template issues.

**Comment by john-m-liu** [comment]
You missed changing a couple spots when you removed ProjectData from the struct

**Comment by john-m-liu** [comment]
Grid changes lgtm

Still need to have the task drawer and perf dashboard pages correctly find the version context

**Pull request merged**